lib/class-wp-rest-themes-controller.php

```diff
@@ -0,0 +1,221 @@
+		$this->rest_base = 'themes';
+	}
+
+	/**
+	 * Registers the routes for the objects of the controller.
+	 *
+	 * @since 5.0.0
+	 *
+	 * @see register_rest_route()
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/active',
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_item' ),
+					'permission_callback' => array( $this, 'get_item_permissions_check' ),
+					'args'                => array(),
+				),
+				'schema' => array( $this, 'get_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Checks if a given request has access to read the theme.
+	 *
+	 * @since 5.0.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has read access for the item, otherwise WP_Error object.
+	 */
+	public function get_item_permissions_check( $request ) {
+		if ( ! is_user_logged_in() || ! current_user_can( 'edit_posts' ) ) {
+			return new WP_Error( 'rest_user_cannot_view', __( 'Sorry, you are not allowed to view themes.', 'gutenberg' ), array( 'status' => rest_authorization_required_code() ) );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Retrieves the active theme.
+	 *
+	 * @since 5.0.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_item( $request ) {
+		$theme = wp_get_theme();
+
+		$response = $this->prepare_item_for_response( $theme, $request );
+		$response = rest_ensure_response( $response );
+
+		return $response;
+	}
+
+	/**
+	 * Prepares a single theme output for response.
+	 *
+	 * @since 5.0.0
+	 *
+	 * @param WP_Theme        $theme   Theme object.
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response Response object.
+	 */
+	public function prepare_item_for_response( $theme, $request ) {
+		$data   = array();
+		$fields = $this->get_fields_for_response( $request );
+
+		if ( in_array( 'name', $fields, true ) ) {
+			$data['name'] = $theme->__get( 'name' );
+		}
+
+		if ( in_array( 'stylesheet', $fields, true ) ) {
+			$data['stylesheet'] = $theme->__get( 'stylesheet' );
+		}
+
+		if ( in_array( 'version', $fields, true ) ) {
+			$data['version'] = $theme->__get( 'version' );
+		}
+
+		if ( in_array( 'template', $fields, true ) ) {
+			$data['template'] = $theme->__get( 'template' );
+		}
+
+		if ( in_array( 'theme_supports', $fields, true ) ) {
+			$formats                           = get_theme_support( 'post-formats' );
+			$formats                           = is_array( $formats ) ? array_values( $formats[0] ) : array();
+			$formats                           = array_merge( array( 'standard' ), $formats );
+			$data['theme_supports']['formats'] = $formats;
+
+			$post_thumbnails = get_theme_support( 'post-thumbnails' );
+
+			if ( $post_thumbnails ) {
```

Comment:
Should we always include `post-thumbnails` in the `theme_supports` and set it to `false`?

Reply:
I’d think so; it’s good to be able to predict the shape of our responses, consistently including properties is useful

Reply:
The changes now ensure `post-thumbnails` is always included in `theme_supports`. The name, etc. fields are now removed.